go/chat/helper.go

```diff
@@ -858,16 +858,43 @@ func newNewConversationHelper(g *globals.Context, uid gregor1.UID, tlfName strin
 	}
 }
 
-func (n *newConversationHelper) findConversations(ctx context.Context,
-	membersType chat1.ConversationMembersType, topicName string, useLocalData bool) ([]chat1.ConversationLocal, error) {
-	onechatpertlf := true
-	return FindConversations(ctx, n.G(), n.DebugLabeler, useLocalData, n.ri, n.uid, n.tlfName, n.topicType,
-		membersType, n.vis, topicName, &onechatpertlf)
+func (n *newConversationHelper) findExisting(ctx context.Context, tlfID chat1.TLFID, topicName string,
+	useLocalData bool) (res []chat1.ConversationLocal, err error) {
+	ib, _, err := n.G().InboxSource.Read(ctx, n.uid, types.ConversationLocalizerBlocking, true, nil,
+		&chat1.GetInboxLocalQuery{
+			Name: &chat1.NameQuery{
+				TlfID:       &tlfID,
+				MembersType: n.membersType,
+			},
+			TlfVisibility: &n.vis,
+			TopicName:     &topicName,
+			TopicType:     &n.topicType,
+		}, nil)
+	if err != nil {
+		return res, err
+	}
+	return ib.Convs, nil
 }
 
-func (n *newConversationHelper) findExisting(ctx context.Context, topicName string, useLocalData bool) (res []chat1.ConversationLocal, err error) {
-	// proceed to findConversations for requested member type
-	return n.findConversations(ctx, n.membersType, topicName, useLocalData)
+func (n *newConversationHelper) getNameInfo(ctx context.Context) (res types.NameInfo, err error) {
+	isPublic := n.vis == keybase1.TLFVisibility_PUBLIC
+	switch n.membersType {
+	case chat1.ConversationMembersType_IMPTEAMUPGRADE:
+		return res, errors.New("cannot create new impteam upgrade conv")
```

Comment:
i think this would cause problems when sending to `zapu@github` for example, this check shouldn't happen unless we don't find an existing conversation

Reply:
This is part of the new conversation path, we never invoke that with impteamupgrade, and if we do it is a mistake.

Reply:
mm true, i'd just double check that `keybase chat send zapu@github test` works with this patch